packages/ag-charts-community/src/chart/legend.ts

```diff
@@ -369,6 +369,15 @@ export class Legend extends BaseProperties {
         this.proxyLegendDirty = false;
     }
 
+    private onDataUpdate(oldData: CategoryLegendDatum[], newData: CategoryLegendDatum[]) {
+        this.proxyLegendDirty =
+            oldData.length !== newData.length ||
+            oldData.some((_v, index, _a) => {
+                const [newValue, oldValue] = [newData[index], oldData[index]];
+                return newValue.id !== oldValue.id;
```

Comment:
> [!CAUTION]
> I do not know if just checking the legend datum ID is sufficient. This condition may need to be tweaked.

Reply:
I think it's probably a reasonable approximation - we need to exercise it in Integrated Charts cases to be sure though.